### PR TITLE
fix(dev): add specific dockerize to development

### PR DESCRIPTION
**Pull request by JellyBellyDev**
[fixup/dockerize-development 44ae65e] fix(dev): add specific dockerize to development

### CODE REVIEW

File: docker-compose.override.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+version: '3.8'
+
+services:
+
+  nginx:
+    image: nginx:alpine
```
Comment (fabiocicerchia) [inline docker-compose.override.yml:6]: I'd suggest using the same images of prod, then customise whatever you like inside of them using the volumes.

File: docker-compose.override.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - ./sys/nginx/dev/default.conf:/etc/nginx/conf.d/default.conf:ro
+      - ./var/log/nginx/:/var/log/nginx:cached
+    ports:
+      - '8001:80'
+
+
+  phpfpm:
+    build:
+      context: .
+      dockerfile: sys/docker/alpine-phpfpm/dev/Dockerfile
```
Comment (fabiocicerchia) [inline docker-compose.override.yml:20]: in this way, we might end up missing critical configurations, and be in one of those classic "it works on my pc" ;)

**Review by fabiocicerchia — CHANGES_REQUESTED** [review]

File: docker-compose.override.yml (hunk 1 above)
Comment (drAlberT) [reply docker-compose.override.yml:6]: totally agreed

File: sys/docker/alpine-phpfpm/dev/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        opcache \
+        zip \
+    && pecl install msgpack yaml redis
+
+
+# REF: https://github.com/api-platform/api-platform/issues/1532
+RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ --allow-untrusted gnu-libiconv
+ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so
+
+COPY --from=composer /usr/bin/composer /usr/local/bin/composer
```
Comment (drAlberT) [inline sys/docker/alpine-phpfpm/dev/Dockerfile:37]: I'd specify the major version (possibly with a BUILD_ARG)

File: sys/docker/alpine-phpfpm/dev/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,39 @@
+FROM php:7.4.11-fpm-alpine3.12
```
Comment (drAlberT) [inline sys/docker/alpine-phpfpm/dev/Dockerfile:1]: I'd add a build arg PHP_VERSION with default value the one you need .. this way to test a different version, in local for example, you don't have to change the file and don't rick to commit it by mistake.

File: sys/docker/alpine-phpfpm/dev/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        gd \
+        intl \
+        mbstring \
+        opcache \
+        zip \
+    && pecl install msgpack yaml redis
+
+
+# REF: https://github.com/api-platform/api-platform/issues/1532
+RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ --allow-untrusted gnu-libiconv
```
Comment (drAlberT) [inline sys/docker/alpine-phpfpm/dev/Dockerfile:34]: ` --allow-untrusted` is ugly ... very ugly

File: sys/nginx/dev/default.conf [hunk 6]
```diff
@@ -0,0 +1,43 @@
+server {
```
Comment (drAlberT) [inline sys/nginx/dev/default.conf:1]: are we sure we need a different NGINX config for local dev? .. we should avoid this IMHO

**Review by drAlberT — CHANGES_REQUESTED** [review]

**Pull request closed without merging**